### PR TITLE
Sea-ice analysis insertion

### LINKED ISSUE #1103: Direct insertion of the seaice analysis
Follow up to #1092 bu in the g-w side.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
# Description
Allows cycling and restarting CICE with the sea-ice analysis if the marine DA is switched on.

Resolves NOAA-EMC/GDASApp#1103

# Type of change
- New feature

# Change characteristics
- Is this a breaking change (a change in existing functionality)? **NO**
- Does this change require a documentation update? **NO**

# How has this been tested?
Cycled ~12 days on Hera.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
make sense.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Comment by guillaumevernieres** [comment]
Arg ... I'm an idiot ... I forgot to add the CICE analysis restart for the first 1/2 cycle. I have not looked at the log yet, but I suspect this is at least part of the problem.

**Comment by JessicaMeixner-NOAA** [comment]
> Experiment C48mx500_3DVarAOWCDA **FAILED** on Hera with error logs:
> 
> ```
> /scratch1/NCEPDEV/global/CI/2584/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_8ded48b2/logs/2021032412/gdasfcst.log
> ```
> 
> Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/8f85e45c86abaad2f6352111b06b2c9f)

If you ls -l this file: 

```
$ ls -ltr /scratch1/NCEPDEV/global/CI/2584/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_8ded48b2/logs/2021032412/
total 324
-rw-r--r-- 1 Terry.McGuinness global 161437 May 10 06:31 gdasfcst.log.0
-rw-r--r-- 1 Terry.McGuinness global 158826 May 10 06:37 gdasfcst.log

```

it doesn't seem like anything has been re-run ?

**Comment by WalterKolczynski-NOAA** [comment]
> > Experiment C48mx500_3DVarAOWCDA **FAILED** on Hera with error logs:
> > ```
> > /scratch1/NCEPDEV/global/CI/2584/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_8ded48b2/logs/2021032412/gdasfcst.log
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/8f85e45c86abaad2f6352111b06b2c9f)
> 
> If you ls -l this file:
> 
> ```
> $ ls -ltr /scratch1/NCEPDEV/global/CI/2584/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_8ded48b2/logs/2021032412/
> total 324
> -rw-r--r-- 1 Terry.McGuinness global 161437 May 10 06:31 gdasfcst.log.0
> -rw-r--r-- 1 Terry.McGuinness global 158826 May 10 06:37 gdasfcst.log
> ```
> 
> it doesn't seem like anything has been re-run ?

We're already looking into it. I think the issue is it jumped right back to running, skipping over the experiment creation. But creation is (for now) where we are doing the staging of cycled ICs.

**Comment by WalterKolczynski-NOAA** [comment]
Okay, I think I know what the problem is. There is no update in this PR in `workflow/setup_expt.py` to copy the ice restart to the correct place in `ROTDIR`.

**Comment by guillaumevernieres** [comment]
> Okay, I think I know what the problem is. There is no update in this PR in `workflow/setup_expt.py` to copy the ice restart to the correct place in `ROTDIR`.

ouch ... A problem for next week. Thanks for looking into this @WalterKolczynski-NOAA .

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh]: This may require a change to gdas.cd, in which case we can defer to later, but we would prefer additional directories levels not be added to the COM templates.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Just the one note about adding `restart/` to the ice analysis path.

File: ush/forecast_postdet.sh
Comment (guillaumevernieres) [reply ush/forecast_postdet.sh]: Yes, good catch again @WalterKolczynski-NOAA . This was discussed offline with @aerorahul and I think we wanted the analysis restart under  ```${COM_ICE_ANALYSIS}/restart/```. 
I can :
- issue the companion pr in the gdasapp with the updated path
- remove the restart directory (I'd rather do that!)

Let me know.

Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh]: I would prefer to get rid of the `restart/` in the path, but I don't know what you and @aerorahul discussed, so I'll wait for him to chime in.

Comment (aerorahul) [reply ush/forecast_postdet.sh]: @guillaumevernieres and @aerorahul discussed this and we agreed on placing the analysis file in `${COM_ICE_ANALYSIS}/`.  It is not really a `restart` since it is an `analysis`!

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Comment by guillaumevernieres** [comment]
I forgot about this:
```
/scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/12/analysis/ice/restart/20210324.090000.cice_model_anl.res.nc
```
needs to be replaced by 
```
/scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/12/analysis/ice/20210324.090000.cice_model_anl.res.nc
```

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres will do that. There are also some other CI issues right now with directories not being cleared when we start over.

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres will do that. There are also some other CI issues right now with directories not being cleared when we start over.

Thanks @WalterKolczynski-NOAA , can you ping me when done? I'll (try to) run the test the same way it is done in the g-w before you re-run the ci.

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres should be done on Hera now

**Comment by guillaumevernieres** [comment]
@WalterKolczynski-NOAA and/or @aerorahul , the link name for the ice analysis IC's isn't correct:
```
/scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/12/analysis/ice/20210324.090000.cice_model.res.nc
```
should be 
```
/scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/gdas.20210324/12/analysis/ice/20210324.090000.cice_model_anl.res.nc
```

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres link has been renamed

**Comment by CatherineThomas-NOAA** [comment]
@guillaumevernieres With the link updated, is this PR good to go?

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres With the link updated, is this PR good to go?

No. I'll ping you here when it's ready @CatherineThomas-NOAA . Working on fixing other unrelated bugs, I'm hopping to get back to this later today.

**Comment by guillaumevernieres** [comment]
Sorry for the delay on this work @WalterKolczynski-NOAA . It's ready to be tested again.

I did rerun the exact same config as the `ci` so hopefully things will work this time around.
my test is here:
```
/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/runs/prs/global-workflow/sorc/gdas.cd/build/gdas/test/gw-ci/C48mx500_3DVarAOWCDA
```
I didn't run every thing, but enough to cover the script changes of this PR (I think ....):
The first 1/ cycle runs fine and the DA + fcst for the full cycle are OK too:
gw-ci-seaiceda

**Comment by RussTreadon-NOAA** [comment]
Any estimate as to when this PR can be merged into develop?

**Comment by WalterKolczynski-NOAA** [comment]
> Any estimate as to when this PR can be merged into develop?

As soon as the CI tests pass.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA quick question, I noticed that the CI for hera is still in "ready" state.  Is there a reason it hasn't started yet?

**Comment by TerrenceMcGuinness-NOAA** [comment]
@JessicaMeixner-NOAA The connection over the internet to the Jenkins Controller had been interrupted and the self restart did clone correctly.  I'm going to restart it now.

**Pull request merged**
